### ISSUE TITLE
Benchmark templating

### ISSUE THREAD

**Issue by rvansa** [Hyperfoil/Hyperfoil#209]
The ability to customize benchmarks with parameters is an often requested feature. So far this has been rejected as there are existing templating systems that are specialized for this job and offer more features than we'd like to include, but the ease of use of any built-in solution is tempting.

The format of the template should be still a valid YAML. Since `${var}` syntax is already used for runtime pattern replacement we can get inspired from existing YAML templating systems that use tags. A simple param replacement could look like

```
foo: !param FOO <optional default value>
```
In case that we want to customize a part of string we could do that using concat:
```
url: !concat [ "http://", !param "SERVER <optional default value>", "/foo" ]
```

The CLI-based workflow: `upload` command should work the same way. Subsequent `run`, though, should ask for params if these are not provided using `-p FOO=bar`. The params provided in last run will stay in CLI context (similar to the name of the last uploaded/executed benchmark) so we don't have to type them again; server should not keep the 'resolved' version of the benchmark.

One downside is that if the benchmark is templated we can't assume that we are able to parse and validate it on upload, only when trying to run it. For templated benchmarks we can only print out which parameters are found (with their default values).

Controller can't keep the serialized version of benchmark but we can still keep it in `/benchmarks` directory.

Templating will be executed as an independent step before the actual parsing (invoking builders' methods) happens. We don't need to duplicate YAML parsing, instead we could adapt the stream of parsing events. Since we're there it might be beneficial to move anchor/alias resolution to that level, too, simplifying the parsing itself.

**Comment by rvansa** [comment]
While we'll be on that I'll probably implement simple loop:
```
foo: !foreach
  items: !param LIST one,two,three
  separator: "," # default separator
  param: ITEM # default param name  
  do:
    bar: !param ITEM
```
that would render
```
foo:
- bar: one
- bar: two
- bar: three
```

**Comment by rvansa** [comment]
Implemented in 5af2a99888ac433ac6791c3daa5c0a37df5c95b8
